src/main/java/software/amazon/nio/spi/s3/S3ClientStore.java

```diff
@@ -34,54 +19,9 @@ public class S3ClientStore {
 
     private static final S3ClientStore instance = new S3ClientStore();
 
-    /**
-     * Default client using the "https://s3.us-east-1.amazonaws.com" endpoint
-     */
-    public static final S3Client DEFAULT_CLIENT = S3Client.builder()
```

Comment:
Removing a public method is a breaking change. I suggest initializing this with the default client of the new S3 provider